### ISSUE TITLE
Lua feature to hide player nametags

### ISSUE THREAD

**Issue by Lemmmy** [minetest/minetest#1345]
A feature to hide a nametag of one player from another
Usage: ```player:set_nametag_visible(otherPlayerName, visible)```
Example: ```playerJim:set_nametag_visible(playerBob, false)```
Player Jim can not see Player Bob's nametag.

Useful in: Games, servers etc
For example, you can hide enemy positions in a game, or in hide and seek
You can hide player nametags (but not players) over distance
Invisibility spells/potions/admin command

**Comment by Wuzzy2** [comment]
> Setting the color with 0 alpha would be equivalent to hiding it.

No, and cheaters would cheer in joy. ;-)
Because the server would still be sending the nametag to the client. A modified client could then easily make those “invisible” nametags visible.

**Comment by Calinou** [comment]
> No, and cheaters would cheer in joy. ;-)
> Because the server would still be sending the nametag to the client. A modified client could then easily make those “invisible” nametags visible.

Nametags are generated by the client, not by the server, AFAIK.

**Comment by Wuzzy2** [comment]
Is the client also informed about *all* player positions, or just of those in proximity?

**Comment by C1ffisme** [comment]
The problem with making nametags have an alpha channel is that there are many glitches with invisibility at the moment.

If it was me, I would add functions called:

```minetest.set_name_hidding(true/false)```
```minetest.set_name_hidding_radius(distance you have to go before the name hides)```

**Comment by Calinou** [comment]
> Is the client also informed about all player positions, or just of those in proximity?

All, unless the server doesn't send far away players (which is currently a server setting).

**Comment by est31** [comment]
@Wuzzy2 thats precisely why the issue is not about showing nametags or not, its about whether to send the coordinates of another player to some client.

**Comment by ProgrammingChicken** [comment]
...I want this feature added, but I would change the method name to:
Template: `player:minetest.set_nametag_visibility(boolean, array)`
array is an optional array of usernames that will be affected, otherwise it will affect all players
soo...
Example:
```
player = "gamerdude"
player:minetest.set_nametag_visibility(true, ["John", "bestfiter", "Jason"])
```
Players John, bestfiter, and Jason would now see Player gamerdude's nametag.
Perhaps the Modding API could have a list for every player:
`nametaggamerdude = [[132, 12, 345],[],["John", "bestfiter", "Jason"]]
But this can't really go without things like:
```
player:minetest.set_nametag(string, array)
player:minetest.set_nametag_color(array, string, array)
player:minetest.set_nametag_view_distance(number, array)
```
#### Key:
##### set_nametag
String is new nametag text.
Array is optional; contains the players that will be affected.
##### set_nametag_color
Array is the color data. (If string is nil then default is RGB type)
String is optional; the type of color system being used.
Second Array is optional; contains the players that will be affected.
##### set_nametag_view_distance
Number is the max viewing distance of nametag.
Array is optional; contains the players that will be affected.

Conclusion:
###### MUST BE ADDED

**Comment by ProgrammingChicken** [comment]
minetest_game use!:
Sneak is not sneaking...
Until you use `minetest.set_nametag_view_distance()` :
Example code:
```
while true do
	for _,user in ipairs(minetest.get_connected_players()) do
		player = user:get_player_name()
		control = player:get_player_control()
		if  control["sneak"] then
			player:minetest.set_nametag_view_distance(1)
		else
			player:minetest.set_nametag_view_distance(nil)
		end
	end
end
```

**Comment by wilkgr76** [comment]
@ProgrammingChicken I like the set_nametag idea; it would make an AFK mod so much easier. This way, it would be possible to change "wilkgr" to "wilkgr (AFK)". 

/me can't wait for this to be added.

**Comment by ProgrammingChicken** [comment]
@wilkgr76 I was thinking about using it for ranks, like ```[n00b] gamerdude``` and ```[Donator] wilkgr```.
/me can't wait either.

**Comment by wilkgr76** [comment]
@ProgrammingChicken Yay! I donated! Just kidding. It's just that I'm really, really annoyed by the lack of a "/away" or "/afk" command.

**Comment by ProgrammingChicken** [comment]
@wilkgr76 ```[AFK] [MODERATOR] Potato360NOSCOPEZ```

**Comment by rubenwardy** [comment]
minetest.set_nametag_attributes is a much better place for this, maybe it should be expanded to allow different settings for each viewer:

Player A sees Player D's tag as green.
Player B sees Player D's tag as red.
Player C can't see Player D's tag.

**Comment by ProgrammingChicken** [comment]
@rubenwardy, i say NU!
there should be rgb codes before text to set its color for that respective section.
aka 
`26FF17[AFK] `24BEFF[MODERATOR] `000000Potato360NOSCOPEZ

**Comment by rubenwardy** [comment]
the ability to add more text to nametags would be good too, but not essential.

**Comment by ProgrammingChicken** [comment]
```
function addToNametag(user, text)
local nametag = get_nametag(user)
user:minetest.set_nametag(nametag..text)
end
```

**Comment by rubenwardy** [comment]
That's not a very good API syntax, we already have player:set_nametag_attributes()

**Comment by LemonLake** [comment]
lmao I created this issue over a year ago and it's still being bumped

**Comment by neoascetic** [comment]
Want this

**Comment by neoascetic** [comment]
@Lemmmy yeah, that what makes minetest minetest

**Comment by Rui-Minetest** [comment]
set_nametag_attributes is implemented.
This issue can be closed.

**Comment by AntumDeluge** [comment]
I created a mod that might be related to this ([hidename][]). Just wanted to mention it in case it is helpful. It allows players to manually hide their nametags from others. As long as the setting *hidename.use_alpha* is enabled, it should be working. I haven't tested thoroughly with it disabled.

(sorry for commented in a topic that was closed so long ago)

[hidename]: https://forum.minetest.net/viewtopic.php?t=18190
